### PR TITLE
Feature icon project

### PR DESCRIPTION
**Pull request by summercode18**
Setup maven module "icon-manager"

### CODE REVIEW

File: open-sphere-base/icon-manager/src/main/java/io/opensphere/icon/manager/IconManagerDialog.java [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+import javafx.scene.control.TextField;
+import javafx.scene.control.ToggleGroup;
+import javafx.scene.control.TreeView;
+import javafx.scene.input.MouseEvent;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.HBox;
+import javafx.scene.text.Text;
+import javafx.stage.Stage;
+
+public class IconManagerDialog
```
Comment (roddyaj) [inline open-sphere-base/icon-manager/src/main/java/io/opensphere/icon/manager/IconManagerDialog.java]: Class comment

File: open-sphere-base/icon-manager/src/main/java/io/opensphere/icon/manager/IconManagerDialog.java [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+{
+    /** The AnchorPane myTopMenuBar; Top window of GUI */
+    @FXML
```
Comment (roddyaj) [inline open-sphere-base/icon-manager/src/main/java/io/opensphere/icon/manager/IconManagerDialog.java]: I'm not too sure about using FXML

File: open-sphere-base/icon-manager/src/main/java/io/opensphere/icon/manager/IconManagerDialog.java [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+    private Text myNotifyText;
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+    /**
+     
+     */
+    private boolean myChoice;
+
+    @FXML
+    void ADD(MouseEvent event)
```
Comment (roddyaj) [inline open-sphere-base/icon-manager/src/main/java/io/opensphere/icon/manager/IconManagerDialog.java]: For these methods and below, method names should start with a lowercase letter, and be camel case after that.

File: open-sphere-base/icon-manager/src/main/java/io/opensphere/icon/manager/IconManagerDialog.java [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+
+    /** The bottom center notification bar. */
+    @FXML
+    private TextField myDataBar;
+
+    /** The bottom left notification text. */
+    @FXML
+    private Text myNotifyText;
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
```
Comment (roddyaj) [inline open-sphere-base/icon-manager/src/main/java/io/opensphere/icon/manager/IconManagerDialog.java]: Dividers like this and the one below aren't very reliable.  They can quickly become irrelevant after someone re-orders the class members, so I'd recommend avoiding them.

File: open-sphere-base/icon-manager/src/main/java/io/opensphere/icon/manager/IconManagerDialog.java [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+
+    @FXML
+    void handleAddIconButtonAction(MouseEvent event)
+    {
+        // myNotifyText.setText("Icon Added to Favorites");
+        // loadFromFile(IconRecord.USER_ADDED_COLLECTION, null);
+    }
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+    private void setIconSize(boolean myChoice)
```
Comment (roddyaj) [inline open-sphere-base/icon-manager/src/main/java/io/opensphere/icon/manager/IconManagerDialog.java]: 'my' prefix should be reserved for member variables, per our coding standards

File: open-sphere-base/icon-manager/src/main/java/io/opensphere/icon/manager/IconManagerFrame.java [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+import java.io.FileInputStream;
+
+import javafx.application.Application;
+import javafx.fxml.FXMLLoader;
+import javafx.scene.Scene;
+import javafx.scene.image.Image;
+import javafx.scene.layout.AnchorPane;
+import javafx.stage.Stage;
+
+public class IconManagerFrame extends Application
```
Comment (roddyaj) [inline open-sphere-base/icon-manager/src/main/java/io/opensphere/icon/manager/IconManagerFrame.java]: I assume this class will eventually go away?  If so lets put a little comment about this being a test driver or something, then we don't need to be as picky when reviewing changes for it.

File: open-sphere-base/icon-manager/src/main/java/io/opensphere/icon/manager/IconManagerFrameTest.fxml [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+<?import javafx.scene.control.TreeView?>
+<?import javafx.scene.layout.AnchorPane?>
+<?import javafx.scene.layout.ColumnConstraints?>
+<?import javafx.scene.layout.GridPane?>
+<?import javafx.scene.layout.HBox?>
+<?import javafx.scene.layout.RowConstraints?>
+<?import javafx.scene.text.Font?>
+<?import javafx.scene.text.Text?>
+
+<AnchorPane minHeight="-Infinity" minWidth="-Infinity"
```
Comment (roddyaj) [inline open-sphere-base/icon-manager/src/main/java/io/opensphere/icon/manager/IconManagerFrameTest.fxml]: Yeah, I'm just not so sure about FXML.  We may want to decide as a team if we're cool using it.

**Comment by rkausch** [comment]
Closing as OBE.

**Pull request closed without merging**